File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) == len(t):
            """dict = {}
            for char in s:
                if char not in dict:
                    dict[char] = 1
                else:
                    dict[char] += 1"""
            dict = Counter(s)
            for key, value in dict.items():
                if t.count(key) != value:
                    return False
        else:
            return False
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        Map<Character, Integer> map = new HashMap<>();
        for (char c: s.toCharArray()) {
            map.put(c, map.getOrDefault(c,0) + 1);
        }
        for (char c: t.toCharArray()) {
            map.put(c, map.getOrDefault(c,0) - 1);
        }
        map.entrySet().removeIf(entry -> entry.getValue() == 0);
        if (map.isEmpty()) return true;
        return false;
    }
}